android/src/main/java/com/polidea/blemulator/bridging/DartMethodCaller.java

```diff
@@ -373,4 +354,97 @@ public void notImplemented() {
+                        Log.e(TAG, "writeCharacteristicForDevice not implemented");
+                    }
+                });
+    }
+
+    public void writeCharacteristicForService(
+            final int serviceIdentifier,
+            final String characteristicUUID,
+            final String valueBase64,
+            final OnSuccessCallback<Characteristic> onSuccessCallback,
+            final OnErrorCallback onErrorCallback) {
+        HashMap<String, Object> arguments = new HashMap<String, Object>() {{
+            put(ArgumentKey.SERVICE_IDENTIFIER, serviceIdentifier);
+            put(ArgumentKey.CHARACTERISTIC_UUID, characteristicUUID);
+            put(ArgumentKey.VALUE, Base64Converter.decode(valueBase64));
+        }};
+        dartMethodChannel.invokeMethod(DartMethodName.WRITE_CHARACTERISTIC_FOR_SERVICE,
+                arguments,
+                new MethodChannel.Result() {
+                    @Override
+                    public void success(@Nullable Object characteristicJsonObject) {
+                        onSuccessCallback.onSuccess(characteristicJsonDecoder.decode((Map<String, Object>) characteristicJsonObject));
+                    }
+
+                    @Override
+                    public void error(String errorCode, @Nullable String message, @Nullable Object bleErrorJsonObject) {
+                        onErrorCallback.onError(new BleError(BleErrorCode.UnknownError, message, 0)); //TODO Add proper error parsing here
+                    }
+
+                    @Override
+                    public void notImplemented() {
+                        Log.e(TAG, "writeCharacteristicForService not implemented");
+                    }
+                });
+    }
+
+    public void writeCharacteristic(
+            final int characteristicIdentifier,
+            final String valueBase64,
+            final OnSuccessCallback<Characteristic> onSuccessCallback,
+            final OnErrorCallback onErrorCallback) {
+        HashMap<String, Object> arguments = new HashMap<String, Object>() {{
+            put(ArgumentKey.CHARACTERISTIC_IDENTIFIER, characteristicIdentifier);
+            put(ArgumentKey.VALUE, Base64Converter.decode(valueBase64));
+        }};
+        dartMethodChannel.invokeMethod(DartMethodName.WRITE_CHARACTERISTIC_FOR_IDENTIFIER,
+                arguments,
+                new MethodChannel.Result() {
+                    @Override
+                    public void success(@Nullable Object characteristicJsonObject) {
+                        onSuccessCallback.onSuccess(characteristicJsonDecoder.decode((Map<String, Object>) characteristicJsonObject));
+                    }
+
+                    @Override
+                    public void error(String errorCode, @Nullable String message, @Nullable Object bleErrorJsonObject) {
+                        onErrorCallback.onError(new BleError(BleErrorCode.UnknownError, message, 0)); //TODO Add proper error parsing here
```

Comment:
please add an issue for the TODO(or link existing one) or implement a proper solution